src/molecule_plugins/podman/playbooks/create.yml

```diff
@@ -144,52 +131,56 @@
-        {% if item.ip is defined %}--ip={{ item.ip }}{% endif %}
-        {% if item.etc_hosts is defined %}
-        {% for i, k in item.etc_hosts.items() %}{% if i != item.name %}--add-host {{ i }}:{{ k }} {% endif %}{% endfor %}{% endif %}
-        {% if item.hostname is defined %}--hostname={{ item.hostname }}{% elif item.name is defined %}--hostname={{ item.name }}{% endif %}
-        {% if item.systemd is defined %}--systemd={{ item.systemd | string | lower }}{% endif %}
-        {{ item.extra_opts | default([]) | join(' ') }}
-        {{ item.pre_build_image | default(false) | ternary('', 'molecule_local/') }}{{ item.image }}
-        {{ (command_directives_dict | default({}))[item.name] | default('') }}
+      containers.podman.podman_container:
+        name: "{{ item.name }}"
+        cap_add: "{{ item.capabilities | default(omit) }}"
+        # it's necessary to exclude empty args as ansible modules run_command method doesn't handle this well
+        cmd_args: "{{ molecule_podman_args | reject('equalto', '') | list + item.extra_opts | default([]) }}"
+        command: "{{ (command_directives_dict | default({}))[item.name] | default('') }}"
+        detach: "{{ item.detach | default(omit) }}"
+        device: "{{ item.devices | default(omit) }}"
+        dns: "{{ item.dns_servers | default(omit) }}"
+        env: "{{ item.env | default(omit) }}"
+        etc_hosts: "{{ item.etc_hosts | default(omit) }}"
+        executable: "{{ podman_exec }}"
+        expose: "{{ item.exposed_ports | default(omit) }}"
+        hostname: "{{ item.hostname | default(omit) }}"
+        image: "{{ item.pre_build_image | default(false) | ternary('', 'molecule_local/') }}{{ item.image }}"
+        ip: "{{ item.ip | default(omit) }}"
+        network: "{{ item.network | default(omit) }}"
+        pid: "{{ item.pid_mode | default(omit) }}"
+        privileged: "{{ item.privileged | default(omit) }}"
+        publish: "{{ item.published_ports | default(omit) }}"
+        security_opt: "{{ item.security_opts | default(omit) }}"
+        systemd: "{{ item.systemd | default(omit) }}"
+        tmpfs: "{{ item.tmpfs | default(omit) }}"
+        tty: "{{ item.tty | default(omit) }}"
+        ulimits: "{{ item.ulimits | default(omit) }}"
+        volume: "{{ item.volumes | default(omit) }}"
+        state: started
       register: server
+      changed_when: false
       with_items: "{{ molecule_yml.platforms }}"
       loop_control:
         label: "{{ item.name }}"
       async: 7200
       poll: 0
-      changed_when: true
+      vars:
+        molecule_podman_args:
+          - "{% if item.cgroup_manager is defined %}--cgroup-manager={{ item.cgroup_manager }}{% endif %}"
+          - >-
+            {% if item.restart_policy is defined %}
+            --restart={{ item.restart_policy }}{% if item.restart_retries is defined %}:{{ item.restart_retries }}{% endif %}
+            {% endif %}"
```

Comment:
It's must be a typo.

I test it locally, if I remove the double quote, it will work.

Reply:
Seems that not tests pass (after removing qotes). Thank you  @zhan9san